crates/provider/src/provider/caller.rs

```diff
@@ -1,47 +1,62 @@
+use super::EthCallParams;
 use crate::ProviderCall;
-use alloy_json_rpc::{RpcParam, RpcReturn};
+use alloy_json_rpc::RpcReturn;
+use alloy_network::Network;
 use alloy_rpc_client::WeakClient;
-use alloy_transport::{RpcError, Transport, TransportErrorKind, TransportResult};
-use std::borrow::Cow;
-
-// TODO: Make `EthCall` specific. Ref: https://github.com/alloy-rs/alloy/pull/788#discussion_r1748862509.
+use alloy_transport::{Transport, TransportErrorKind, TransportResult};
 
 /// Trait that helpes convert `EthCall` into a `ProviderCall`.
-pub trait Caller<T, Params, Resp>: Send + Sync
+pub trait Caller<T, N, Resp>: Send + Sync
 where
     T: Transport + Clone,
-    Params: RpcParam,
+    N: Network,
     Resp: RpcReturn,
 {
     /// Method that needs to be implemented to convert to a `ProviderCall`.
     ///
-    /// This method handles serialization of the params and sends the request to relevant data
-    /// source and returns a `ProviderCall`.
+    /// This method sends the request to relevant data source and returns a `ProviderCall`.
     fn call(
```

Comment:
@yash-atreya could we make `Caller::call` always output `Bytes` and `estimate_gas` would output u64? `Resp` generic might be tricky to deal with in custom implementations